### PR TITLE
[CPL][SYSDM][NTDLL] Plan B: Delete ReportAsWorkstation value

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Don't use `ReportAsWorkstation` registry value.
JIRA issue: [CORE-13795](https://jira.reactos.org/browse/CORE-13795)

AFTER:
after

### CODE REVIEW

**Comment by katahiromz** [comment]
See also Plan A: https://github.com/reactos/reactos/pull/2326

File: dll/cpl/sysdm/advanced.c [hunk 1]
```diff
@@ -11,76 +11,153 @@
...
+
+    error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, ProductOptions, 0, KEY_READ, &hKey);
+    if (error)
+        return FALSE;
+
+    cbValue = sizeof(szValue);
+    error = RegQueryValueExW(hKey, L"ProductType", NULL, NULL, (LPBYTE)szValue, &cbValue);
+    if (!error)
+    {
+        if (lstrcmpW(szValue, L"WinNT") == 0)
```
Comment (learn-more) [inline dll/cpl/sysdm/advanced.c]: `lstrcmpiW` is a bit friendlier for manual string entry

File: dll/cpl/sysdm/advanced.c [hunk 2]
```diff
@@ -11,76 +11,153 @@
...
-                       ReportAsWorkstationKey,
-                       0,
-                       NULL,
-                       0,
-                       KEY_WRITE,
-                       NULL,
-                       &hKey,
-                       NULL) == ERROR_SUCCESS)
+    LONG error;
+    WCHAR szValue[32];
```
Comment (learn-more) [inline dll/cpl/sysdm/advanced.c]:
```suggestion
    WCHAR szValue[32] = {0};
```

File: dll/ntdll/rtl/version.c [hunk 3]
```diff
@@ -197,8 +146,14 @@ RtlGetVersion(IN OUT PRTL_OSVERSIONINFOW lpVersionInformation)
         InfoEx->wProductType = SharedUserData->NtProductType;
         InfoEx->wReserved = 0;
 
-        /* HACK: ReactOS specific changes, see bug-reports CORE-6611 and CORE-4620 (aka. #5003) */
-        SetRosSpecificInfo(InfoEx);
+        if (InfoEx->wProductType == VER_NT_SERVER)
```
Comment (learn-more) [inline dll/ntdll/rtl/version.c]: We might want to keep a comment that this is reactos specific?

File: dll/cpl/sysdm/advanced.c (hunk 1 above)
Comment (katahiromz) [reply dll/cpl/sysdm/advanced.c]: OK.

File: dll/cpl/sysdm/advanced.c (hunk 2 above)
Comment (katahiromz) [reply dll/cpl/sysdm/advanced.c]: OK but I don't agree because it slows down the speed.

File: dll/ntdll/rtl/version.c (hunk 3 above)
Comment (katahiromz) [reply dll/ntdll/rtl/version.c]: OK.

**Comment by katahiromz** [comment]
@ColinFinck We have to choose either Plan A or Plan B.

**Comment by HBelusca** [comment]
I would prefer plan A

**Comment by learn-more** [comment]
I would prefer plan B

**Comment by HBelusca** [comment]
OK since people "voted" to my comment I will write a separate one here:
I was thinking actually about having a way to **temporarily** report as workstation or server either globally or just for a given application (would then be via the compatibility settings), but in a temporary way that is not persistent across reboots, and this may or may not be implemented using the "ReportAsWorkstation" trick we used so far.
Separately, a persistent (across reboots) option that changes the reported flavour of the OS could also be added, and this appears to be the so-called Plan-A.

**Comment by ColinFinck** [comment]
As I have been pinged in this PR: I believe that supporting both a "ReactOS Workstation" and a "ReactOS Server" version is more than we can manage. That would result in bug reports depending on what setting was chosen during setup or afterwards. We would have to double testing efforts, because applications could behave differently when they detect a server OS rather than a workstation OS (they already do).

Therefore, my proposed solution looks like this:
* Set a reasonable system-wide default, with no way to change that default system-wide!
And that default should most likely be "Workstation": Given the target audience of ReactOS, it's more likely that people run workstation applications rather than server ones. At least, this is what I get from our bug reports.
* The few cases where the NT Product Type is expected to be "Server" should be handled on a per-application basis via our Application Compatibility Framework. This can also target the few cases where applications may treat everything NT 5.1 as "workstation" and everything NT 5.2 as "server" (although there has been an official NT 5.2 workstation OS in the form of Windows XP x64 Edition).

I think this mostly aligns with what @HBelusca suggested, except that I would object to a separate persistent setting when we already have the Application Compatibility Framework.

**Comment by blackcrack** [comment]
The where not need more as a Workstation at home should it able to install and just to use. The most Peoples need a Workstation at home. The Technicians have the possible to configuration and setting up a Server and know how make it to a server in the configuration.

**Pull request closed without merging**